### DISCUSSION TITLE
Word . (dot) display radix.

### DISCUSSION THREAD

**Discussion by SirWumpus** [ForthHub/discussion#123] [Q&A] [answered]
Concerning Forth 200x Draft 19.1 sections 3.2.1.3 Free-field number display and 6.1.0180 . "dot" neither section make any mention of whether `.` output applies the current `BASE`.  Should `.` apply `BASE` or given that the stack argument being a signed integer `n` imply base 10 output? 

For example I can see a couple options (assume 32 bits):

ok DECIMAL #-1 .
-1 ok
ok HEX #-1 .
-1   ( or )  -FFFFFFFF  ( or )  FFFFFFFF
ok HEX #-1234
-4D2 ( or ) -FFFFFFFFFFFFFB2E ( or ) FFFFFFFFFFFFFB2E

With `U.` I would expect `BASE` to be used and no concern of sign, but in the case of `.` it seems unclear.

**Comment by ruv** [comment, marked as answer]
> Should `.` apply `BASE` 

Yes.

[3.2.1.2 Digit conversion](https://forth-standard.org/standard/usage#subsubsection.3.2.1.2) says:

「 Conversion between the internal and external forms of a digit shall behave as follows:
The value in [BASE](https://forth-standard.org/standard/core/BASE) is the radix for number conversion. 」

So `BASE` is always used by default on both input and output.  And Dot `.` uses the value from `BASE` too.

> or given that the stack argument being a signed integer `n` imply base 10 output?

No. A signed integer just means that it's displayed with leading minus if it's negative.

[3.2.1.3 Free-field number display](https://forth-standard.org/standard/usage#usage:dot) says:
「 If the number is negative, a leading minus sign is displayed 」

NB: an unsigned integer cannot be negative.

Whether a number is interpreted as signed or unsigned is defined by the operation.

So `HEX #-1234 .` shall print either `-4D2 ` or `-4d2 `, since Dot interprets the number as signed.

`-FFFFFFFFFFFFFB2E` is out of the range of 32-bit integers. `FFFFFFFFFFFFFB2E` doesn't have leading minus (that shall be shown on output a singed negative integer).

**Reply by SirWumpus** [reply]
Oops.  Sorry for confusion/  My last example was copy/paste from 64-bit machine.

**Reply by ruv** [reply]
In another example, `-FFFFFFFF` is out of the range of 32-bit integers too, and  `FFFFFFFF` doesn't have leading minus.
